test/unit/src/utils/keyUtils.test.ts

```diff
@@ -1,39 +1,190 @@
+import { describe, expect, it } from '@jest/globals'
 import { keyListAsLeveledKeys } from '../../../../src/utils/keyUtils'
 import { DevSecurityLevel } from '@shardeum-foundation/core/dist/shardus/shardus-types'
 
-describe('keyUtils utility functions', () => {
+/**
+ * Test suite for keyUtils.ts
+ *
+ * This test suite validates the functionality of utility functions
+ * that handle key transformations and manipulations.
+ */
+describe('Key Utilities', () => {
+  /**
+   * Tests for keyListAsLeveledKeys function
+   *
+   * This function transforms an array of string keys into an object
+   * where each key is mapped to a security level value.
+   */
   describe('keyListAsLeveledKeys', () => {
-    it('should convert empty array to empty object', () => {
-      const result = keyListAsLeveledKeys([], DevSecurityLevel.High)
+    /**
+     * Tests the successful transformation of a key list to a leveled keys object
+     */
+    it('should transform a list of keys into an object with security levels', () => {
+      // Arrange
+      const keyList = ['key1', 'key2', 'key3']
+      const securityLevel = 2
+
+      // Act
+      const result = keyListAsLeveledKeys(keyList, securityLevel)
+
+      // Assert
+      expect(result).toEqual({
+        key1: 2,
+        key2: 2,
+        key3: 2,
+      })
+    })
+
+    /**
+     * Tests that different security levels can be assigned
+     */
+    it('should assign the specified security level to each key', () => {
+      // Arrange
+      const keyList = ['keyA', 'keyB']
+      const securityLevel = 5
+
+      // Act
+      const result = keyListAsLeveledKeys(keyList, securityLevel)
```

Comment:
This shouldn't be possible right? we have 0, 1, 2, 3 as values in `DevSecurityLevel` enum? 
This is debatable, but if we are not supporting a certain level in our enum, then we shouldn't be able to assign it.
It's possible that it's prevented somewhere early in the flow and this util never receives 5 as security level. 
My suggestion would be comment this out with a note that, this needs to be handled in main business logic if not done already. 
cc @BelfordZ @afostr